### PR TITLE
feat(camera_web)!: use flutter camera platform interface

### PR DESCRIPTION
**Pull request by bselwe**
## Description

- Replaced photobooth's platform interface with the flutter [camera plugin platform interface](https://github.com/flutter/plugins/blob/master/packages/camera/camera_platform_interface/lib/src/platform_interface/camera_platform.dart#L26) in `camera_web`.
- Removed [`Camera`](https://github.com/VGVentures/photobooth/compare/main...VGVentures:feat/use-flutter-camera-platform-interface#diff-95b0d214cebd76a613f66fbe94ad5a3cbccd21e0084d34b83b8589d2a14b5ed3L109) class temporarily as it depends on the photobooth's platform interface.
- The web camera plugin now throws an unimplemented error for all methods.

## Type of Change

- [ ] ✨ New feature (non-breaking change which adds functionality)
- [ ] 🛠️ Bug fix (non-breaking change which fixes an issue)
- [x] ❌ Breaking change (fix or feature that would cause existing functionality to change)
- [ ] 🧹 Code refactor
- [ ] ✅ Build configuration change
- [ ] 📝 Documentation
- [ ] 🗑️ Chore

### CODE REVIEW

File: packages/camera/camera_web/lib/src/camera_web.dart [hunk 1]
```diff
@@ -18,213 +16,170 @@ class CameraPlugin extends CameraPlatform {
...
-  Future<void> play(int textureId) => _cameras[textureId]!.play();
+  Stream<VideoRecordedEvent> onVideoRecordedEvent(int cameraId) {
+    throw UnimplementedError('onVideoRecordedEvent() is not implemented.');
+  }
 
   @override
-  Future<void> stop(int textureId) async => _cameras[textureId]!.stop();
+  Stream<DeviceOrientationChangedEvent> onDeviceOrientationChanged() {
+    throw UnimplementedError(
+        'onDeviceOrientationChanged() is not implemented.');
```
Comment (felangel) [inline packages/camera/camera_web/lib/src/camera_web.dart]:
```suggestion
        'onDeviceOrientationChanged() is not implemented.',);
```
nit

File: packages/camera/camera_web/lib/src/camera_web.dart [hunk 2]
```diff
@@ -18,213 +16,170 @@ class CameraPlugin extends CameraPlatform {
...
+  }
 
   @override
-  Future<CameraImage> takePicture(int textureId) {
-    return _cameras[textureId]!.takePicture();
+  Future<void> lockCaptureOrientation(
+      int cameraId, DeviceOrientation orientation) {
```
Comment (felangel) [inline packages/camera/camera_web/lib/src/camera_web.dart]:
```suggestion
      int cameraId, DeviceOrientation orientation,) {
```

File: packages/camera/camera_web/lib/src/camera_web.dart [hunk 3]
```diff
@@ -1,14 +1,12 @@
 import 'dart:async';
 import 'dart:html' as html;
-import 'dart:ui' as ui;
+import 'dart:math';
```
Comment (felangel) [inline packages/camera/camera_web/lib/src/camera_web.dart:3]: why do we need the math import?

Comment (bselwe) [reply packages/camera/camera_web/lib/src/camera_web.dart:3]: It is used for the point in [`setExposurePoint`](https://github.com/VGVentures/photobooth/pull/1/files#diff-95b0d214cebd76a613f66fbe94ad5a3cbccd21e0084d34b83b8589d2a14b5ed3R130) and [`setFocusPoint`](https://github.com/VGVentures/photobooth/pull/1/files#diff-95b0d214cebd76a613f66fbe94ad5a3cbccd21e0084d34b83b8589d2a14b5ed3R160).

**Review by felangel — APPROVED** [review]

**Comment by bselwe** [comment]
Moved to https://github.com/flutter/plugins/pull/4151.

**Pull request closed without merging**